src/pages/payment/mobile/mobile.tsx

```diff
@@ -1,19 +1,230 @@
-import { ChangeEvent, useState } from 'react';
-import { Button } from '../../../components/atoms/button/button';
-import { Input } from '../../../components/atoms/input/input';
-import css from './mobile.module.scss';
+import { useState } from "react";
+import { useMatch } from "@tanstack/react-location";
+import Web3 from "web3";
+import { useNavigate } from "@tanstack/react-location";
+
+//COMPONENTS
+import { Header } from "src/components/atoms/header/header";
+import { Button } from "src/components/atoms/button/button";
+import JobDescrioptionCard from "src/components/templates/job-description-card";
+import PaymentSummaryCard from "src/components/templates/payment-summary-card";
+import PaymentMethods from "src/components/templates/payment-methods";
+
+//CONSTANTS
+import config from "../crypto-config";
+
+//API
+import { confirmPayment } from "./mobile.service";
+
+//TYPES
+//FIXME: create a general Resolver types to use in every page
+import { Resolver } from "src/pages/offer-received/offer-received.types";
+
+//STYLES
+import css from "./mobile.module.scss";
 
 export const Mobile = (): JSX.Element => {
-  const [amount, setAmount] = useState('');
+  const navigate = useNavigate();
+  const { offer } = useMatch().ownData as Resolver;
+  const offerId = offer.id;
+  const {
+    job_category: { name: job_name },
+    recipient: {
+      meta: { name: applicant_name, avatar, city, country },
+      type,
+    },
+    project: { payment_scheme },
+    total_hours,
+    assignment_total,
+  } = offer;
+  //FIXME: if no contribute not allow to continue flow
+  const contributor = offer.recipient.meta.wallet_address;
+  const commision = assignment_total * 0.03;
+  const total_price = commision + assignment_total;
+  const amount = "2";
+  const [account, setAccount] = useState("");
+  const [process, setProcess] = useState(false);
 
-  function onAmountChange(e: ChangeEvent<HTMLInputElement>) {
-    setAmount(e.target.value);
+  const web3 = new Web3(window.ethereum);
+  web3.eth.getAccounts().then((accounts) => {
+    setAccount(accounts[0]);
+    web3.eth.defaultAccount = accounts[0];
+  });
+
+  async function connectWallet() {
+    await window.ethereum.enable();
+    const accounts = await web3.eth.getAccounts();
+    setAccount(accounts[0]);
+    web3.eth.defaultAccount = accounts[0];
   }
 
+  async function allowance(escrowAmount: number) {
+    // we may configure this fee ratio later
+    const fee = escrowAmount * 0.03;
+    const allowanceAmount = escrowAmount + fee;
+    const erc20Contract = new web3.eth.Contract(
+      config.token.abi,
+      config.token.address
+    );
+
+    const approved = await erc20Contract.methods
+      .approve(
+        config.escrow.address,
+        web3.utils.toWei(allowanceAmount.toString())
+      )
+      .send({ from: web3.eth.defaultAccount });
+    if (!approved) throw new Error("Allowance not approved for escorw");
+  }
+
+  async function escorw(escrowAmount: number) {
```

Comment:
I suggest moving logic out of the view component to keep data and view separate. You can create a custom data hook for this component to separate the two concerns. This enables better readability and testability of the code.

Here is a small [article](https://felixgerschau.com/react-hooks-separation-of-concerns/) on the subject. You can also find more linked articles at the bottom of the post.

Here is a more advanced example on the topic of separation of concern: [controller](https://github.com/cmdo/valkyr/blob/main/services/app/src/modules/database/views/tests.controller.ts) [view](https://github.com/cmdo/valkyr/blob/main/services/app/src/modules/database/views/tests.view.tsx)